chore: fix Checkbox indeterminate state (#7780)
primer/react: .changeset/proud-bears-study.md

---
"@primer/react": patch
---

> chore: fixes bug in Checkbox indeterminate state where indeterminate state was being lost on checkbox click if the state didn't change

Reviewer: ```suggestion
fix: Checkbox indeterminate state now persists on click if the state didn't change
```

Super small nit, the conventional commit for this would be `fix` versus `chore`
